packages/eslint-plugin/tests/rules/no-extra-semi.test.ts

```diff
@@ -0,0 +1,321 @@
+import rule from '../../src/rules/no-extra-semi';
+import { RuleTester } from '../RuleTester';
+
+const ruleTester = new RuleTester({
+  parser: '@typescript-eslint/parser',
+});
+
+ruleTester.run('no-extra-semi', rule, {
+  valid: [
+    {
+      code: 'var x = 5;',
+    },
+    {
+      code: 'function foo() {}',
+    },
+    {
+      code: 'for(;;);',
+    },
+    {
+      code: 'while(0);',
+    },
+    {
+      code: 'do;while(0);',
+    },
+    {
+      code: 'for(a in b);',
+    },
+    {
+      code: 'for(a of b);',
+      parserOptions: { ecmaVersion: 6 },
+    },
+    {
+      code: 'if(true);',
+    },
+    {
+      code: 'if(true); else;',
+    },
+    {
+      code: 'foo: ;',
+    },
+    {
+      code: 'with(foo);',
+    },
+
+    // Class body.
+    {
+      code: 'class A { }',
+      parserOptions: { ecmaVersion: 6 },
+    },
+    {
+      code: 'var A = class { };',
+      parserOptions: { ecmaVersion: 6 },
+    },
+    {
+      code: 'class A { a() { this; } }',
+      parserOptions: { ecmaVersion: 6 },
+    },
+    {
+      code: 'var A = class { a() { this; } };',
+      parserOptions: { ecmaVersion: 6 },
+    },
+    {
+      code: 'class A { } a;',
+      parserOptions: { ecmaVersion: 6 },
+    },
+
+    // modules
+    {
+      code: 'export const x = 42;',
+      parserOptions: {
+        ecmaVersion: 6,
+        sourceType: 'module',
+      },
+    },
+    {
+      code: 'export default 42;',
+      parserOptions: {
+        ecmaVersion: 6,
+        sourceType: 'module',
+      },
+    },
+
+    // Class Property
+    {
+      code: `
+export class Foo {
+  public foo: number = 0;
+}
+      `,
+    },
```

Comment:
Could you please also add a sanity test for:
```ts
class Foo
  public foo: number = 0; public bar: number = 1;
}
```
Just want to make sure it doesn't remove the semi in that case (i'm certain it won't, just to ensure there's no regression)